src/managers/connect.py

```diff
@@ -177,12 +238,36 @@ def remove_plugin(self, path_prefix: str) -> None:
 
     def ping_connect_api(self) -> requests.Response:
         """Makes a GET request to the unit's Connect API Endpoint and returns the response."""
-        auth = HTTPBasicAuth(
-            self.context.peer_workers.ADMIN_USERNAME, self.context.peer_workers.admin_password
-        )
-        return requests.get(
-            self.context.rest_uri, timeout=self.REQUEST_TIMEOUT, auth=auth, verify=False
-        )
+        return self._request("GET", "/")
+
+    def connector_status(self, relation_id: int) -> TaskStatus:
+        """Returns the managed connector status for given `relation_id`."""
+        for connector, status in self.connectors.items():
+            if re.match(self._managed_connector_regex(relation_id), connector):
+                return status
+
+        return TaskStatus.UNKNOWN
+
+    def stop_connector(self, relation_id: int) -> None:
+        """Stops the managed connector instance for given `relation_id`."""
+        for connector, status in self.connectors.items():
+            if not re.match(self._managed_connector_regex(relation_id), connector):
+                continue
+
+            if status == TaskStatus.STOPPED:
+                logger.debug("Connector is already stopped.")
+                return
+
+            resp = self._request("PUT", f"connectors/{connector}/stop")
+
+            if resp.status_code == 204:
+                logger.debug(f"Successfully stopped connector for relation ID={relation_id}.")
+                return
+
+            logger.error(f"Unable to stop connector, details: {resp.content}")
+            return
+
+        logger.error(f"Unable to find a managed connector for relation ID={relation_id}")
```

Comment:
Nice! In theory, this should handle if we have 1+ connector tasks on the same Connect.